### PR TITLE
chore: resolve linting errors, make linting checks in CI more strict (DEV-2080)

### PR DESCRIPTION
**Pull request by jnussbaum**
resolves DEV-2080

### CODE REVIEW

File: .github/workflows/tests-on-push.yml [hunk 1]
```diff
@@ -2,67 +2,79 @@
...
+        run: poetry run mypy .
+
       - name: Lint Code Base
-        uses: github/super-linter/slim@v4       # GitHub's super-linter: https://github.com/github/super-linter
+        uses: github/super-linter/slim@v4           
+        # GitHub's super-linter: https://github.com/github/super-linter
+        # super-linter uses a pre-built image with all linters in it. Thus, the linters run outside the poetry virtualenv. 
+        # So, pylint reports import-errors for dependencies installed with poetry.
+        # In addition, pylint and mypy don't recognize the ignore patterns defined in pyproject.toml.
+        # For these reasons, pylint and mypy must be run independently from super-linter.
```
Comment (jnussbaum) [inline .github/workflows/tests-on-push.yml:51]: This is the only substantial change in this file

File: .yaml-lint.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
+---
```
Comment (jnussbaum) [inline .yaml-lint.yml:1]: I copy-pasted this file from the super-linter templates, and adapted the max. line lenght

File: pyproject.toml [hunk 3]
```diff
@@ -79,16 +82,69 @@ addopts = [
...
 multi_line_output = 3
 known_first_party = ["dsp_tools"]
+
+
+[tool.pylint.MASTER]
+ignore-paths = [
+    "src/dsp_tools/models/.*$",           # TODO: activate this
+    "src/dsp_tools/import_scripts/.*$",   # TODO: activate this
+    "test/unittests/.*$",                 # TODO: activate this
+    "test/e2e/.*$"                        # TODO: activate this
```
Comment (jnussbaum) [inline pyproject.toml:111]: These are ignored for the moment. The long term goal is to activate them one day.

File: src/dsp_tools/utils/project_create.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def _create_project_on_server(
     try:
         # the normal, expected case is that this try block fails
         project_local = Project(con=con, shortcode=project_definition["project"]["shortcode"])
-        project_remote: Project = try_network_action(lambda: project_local.read())
```
Comment (jnussbaum) [inline src/dsp_tools/utils/project_create.py:49]: it was necessary to refactor `try_network_action()` because of https://pylint.pycqa.org/en/latest/user_guide/messages/warning/cell-var-from-loop.html.
In the case of `try_network_action()`, I don't think there was ever a danger, but I think it's important for me to be warned of this potentially dangerous error. 
So I found it necessary to refactor all occurrences where pylint warns me of this potential error, so that if I receive such a warning in the future, I know that I have to take it seriously.

File: src/dsp_tools/utils/xml_upload.py [hunk 5]
```diff
@@ -334,14 +335,21 @@ def _parse_xml_file(input_file: Union[str, Path, etree._ElementTree[Any]]) -> et
...
+    
+    # remove namespace URI from the elements' names and transform the special tags to their technically correct form
     for elem in tree.iter():
-        if isinstance(elem, etree._Comment) or isinstance(elem, etree._ProcessingInstruction):
-            # properties that are commented out would break the the constructor of the class XMLProperty, if they are not removed here.
-            elem.getparent().remove(elem)  # type: ignore
-        else:
-            elem.tag = etree.QName(elem).localname   # remove namespace URI in the element's name
-        
+        elem.tag = etree.QName(elem).localname   # remove namespace URI in the element's name
```
Comment (jnussbaum) [inline src/dsp_tools/utils/xml_upload.py:352]: Semantically, nothing has changed

**Comment by jnussbaum** [comment]
@irinaschubert @BalduinLandolt 
Most of this PR consists of semantically meaningless changes. But the following files contain "real" changes:
 - .github/workflows/tests-on-push.yml
 - pyproject.toml

File: src/dsp_tools/utils/excel_to_json_lists.py [hunk 6]
```diff
@@ -335,7 +335,7 @@ def excel2lists(
     excel_file_paths = _extract_excel_file_paths(excelfolder)
     if verbose:
         print("The following Excel files will be processed:")
-        [print(f" - {filename}") for filename in excel_file_paths]
+        print(*(f" - {filename}" for filename in excel_file_paths), sep="\n")
```
Comment (BalduinLandolt) [inline src/dsp_tools/utils/excel_to_json_lists.py:338]:
```suggestion
        print("\n".join((f" - {filename}" for filename in excel_file_paths)))
```
seems more readable to me

File: src/dsp_tools/utils/excel_to_json_properties.py [hunk 7]
```diff
@@ -159,18 +160,19 @@ def excel2properties(excelfile: str, path_to_output_file: Optional[str] = None)
...
+    if any(df.get(lang) is not None for lang in languages):
         warnings.warn(f"The file '{excelfile}' uses {languages} as column titles, which is deprecated. "
                       f"Please use {[f'label_{lang}' for lang in languages]}")
     if df.get("hlist"):
         warnings.warn(f"The file '{excelfile}' has a column 'hlist', which is deprecated. "
                       f"Please use the column 'gui_attributes' for the attribute 'hlist'.")
 
     # transform every row into a property
-    props = [_row2prop(row, i, excelfile) for i, row in df.iterrows()]
+    props = [_row2prop(row, int(str(i)), excelfile) for i, row in df.iterrows()]
```
Comment (BalduinLandolt) [inline src/dsp_tools/utils/excel_to_json_properties.py:175]: why is that needed?

File: mkdocs.yml [hunk 8]
```diff
@@ -1,9 +1,15 @@
+---
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply mkdocs.yml:1]: Apparently, the yaml syntax defines that every file must start with the 3 dashes. Otherwise, the yamllinter wouldn't have told me to add them, I guess.

File: mkdocs.yml [hunk 9]
```diff
@@ -1,9 +1,15 @@
...
 site_url: https://docs.dasch.swiss/latest/DSP-TOOLS
 repo_url: https://github.com/dasch-swiss/dsp-tools
 repo_name: dasch-swiss/dsp-tools
 edit_uri: blob/main/docs/
-copyright: "&copy; 2017 - <script>document.write(new Date().getFullYear())</script> <a href='https://dasch.swiss' target='_blank'>Swiss National Data and Service Center for the Humanities (DaSCH)</a>"
+copyright: >- 
+  &copy; 2017 - <script>document.write(new Date().getFullYear())</script> 
+  <a href='https://dasch.swiss' target='_blank'>
+  Swiss National Data and Service Center for the Humanities (DaSCH)
+  </a>
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply mkdocs.yml:12]: Thanks for pointing out. I realized that this copyright note is not even visible on docs.dasch.swiss. It is only visible when the docs are built locally. So I removed it.

File: src/dsp_tools/excel2xml.py [hunk 10]
```diff
@@ -345,35 +345,35 @@ def append_permissions(root_element: etree.Element) -> etree._Element:  # type:
...
+    root_element.append(prop_restricted)
 
     return root_element
 
 
 def make_resource(
     label: str,
     restype: str,
-    id: str,
+    id: str,                                     # pylint: disable=redefined-builtin
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply src/dsp_tools/excel2xml.py:376]: I'm reluctant to change the signature of a method that RDU people are using and already accustomed to...

File: src/dsp_tools/utils/shared.py [hunk 11]
```diff
@@ -48,7 +48,11 @@ def login(server: str, user: str, password: str) -> Connection:
     return con
 
 
-def try_network_action(action: Callable[..., Any]) -> Any:
+def try_network_action(
+    action: Callable[..., Any],
+    *args: Any,
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply src/dsp_tools/utils/shared.py:53]: 1) There are several places where I use the positional args, e.g. in project_create_lists.py, line 48 😃 

2) The only possible type annotation of `*args` and `**kwargs` is the type of ONE such argument. It is not possible to annotate them as list/sequence/tuple. See here: https://stackoverflow.com/questions/37031928/type-annotations-for-args-and-kwargs#37032111

**Comment by jnussbaum** [comment]
> In this context it might be worth pointing out [Ruff Linter](https://github.com/charliermarsh/ruff) again - it seems to gain tons of adaption in the python community, lately. (Libraries like FastAPI, ScyPy and Pandas use it) again. I'm not sure how exactly it would fit in your setup best, but it seems pretty cool so maybe worth looking into?

Thanks for the hint, I haven't forgotten it, and I will look into it. For the present PR, I worked with pylint because this is already configured in my IDE, and over time, I began investigating pylint's error messages more closely. I have then read the explanations of the single pylint error messages, and have learnt a lot from them. I began then to experiment how I could resolve the pylint errors, and which of them I have to silence. So for the moment, I decided to just stick with pylint, but I havent forgotten ruff 😄

**Comment by BalduinLandolt** [comment]
> > In this context it might be worth pointing out [Ruff Linter](https://github.com/charliermarsh/ruff) again - it seems to gain tons of adaption in the python community, lately. (Libraries like FastAPI, ScyPy and Pandas use it) again. I'm not sure how exactly it would fit in your setup best, but it seems pretty cool so maybe worth looking into?
> 
> Thanks for the hint, I haven't forgotten it, and I will look into it. For the present PR, I worked with pylint because this is already configured in my IDE, and over time, I began investigating pylint's error messages more closely. I have then read the explanations of the single pylint error messages, and have learnt a lot from them. I began then to experiment how I could resolve the pylint errors, and which of them I have to silence. So for the moment, I decided to just stick with pylint, but I havent forgotten ruff 😄

👍   
As far as I know, ruff implements rules from other linters, so you could start with activating the pylint rules, and the behaviour should be the same.

File: mkdocs.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply mkdocs.yml]: I wonder though, if we should have that on docs.dasch.swiss (ideally with CC0)? Maybe you can bring this up in an RDU meeting?

File: mkdocs.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply mkdocs.yml:1]: weird specification... but ok, fair enough :)

File: src/dsp_tools/excel2xml.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply src/dsp_tools/excel2xml.py:376]: but shadowing built-ins like this should never be done, so it would be an improvement for RDU to not do this too

**Review by irinaschubert — APPROVED** [review]
LGTM

File: mkdocs.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (jnussbaum) [reply mkdocs.yml]: I just brought it up in the RDU chat

File: .github/workflows/release-please.yml [hunk 12]
```diff
@@ -18,5 +18,12 @@ jobs:
           token: ${{ secrets.GH_TOKEN }}
           release-type: python
           package-name: dsp-tools
-          changelog-types: '[{"type": "feat", "section": "Enhancements", "hidden": false }, {"type": "fix", "section": "Bug Fixes", "hidden": false }, {"type": "chore", "section": "Maintenance", "hidden": false }, {"type": "refactor", "section": "Maintenance", "hidden": false }, {"type": "docs", "section": "Documentation", "hidden": false }]'
+          changelog-types: >-
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply .github/workflows/release-please.yml:21]: I learnt about it here: https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines#21699210
